bc-backend/src/main/java/org/jboss/da/bc/backend/impl/FinalizerImpl.java

```diff
@@ -82,35 +81,44 @@ public Integer createBCs(int id, ProjectHiearchy toplevelBc) throws Communicatio
      *      b) multiple integers, when the hiearchy object is not selected AND it has selected dependencies
      *      c) NO integer, when the hiearchy object is not selected AND it has NO selected dependencies
      */
-    Set<Integer> create(ProjectHiearchy hiearchy, Set<Integer> allDependencyIds) throws CommunicationException, PNCRequestException {
+    Set<Integer> create(ProjectHiearchy hiearchy, Set<Integer> allDependencyIds)
+            throws CommunicationException, PNCRequestException {
         Set<Integer> nextLevelDependencyIds = new HashSet<>();
 
         for (ProjectHiearchy dep : hiearchy.getDependencies()) {
             nextLevelDependencyIds.addAll(create(dep, allDependencyIds));
         }
 
-        if(hiearchy.isSelected()){
+        if (hiearchy.isSelected()) {
             BuildConfiguration bc;
             ProjectDetail project = hiearchy.getProject();
             if (project.isUseExistingBc()) {
-                Optional<BuildConfiguration> optionalBc = bcFinder.lookupBcByScm(project.getScmUrl(), project.getScmRevision());
-                bc = optionalBc.orElseThrow(() -> new IllegalStateException("useExistingBC is true, but there is no BC to use."));
+                List<Integer> existingBCIds = bcFinder.lookupBcIdsByScm(project.getScmUrl(),
+                        project.getScmRevision());
+                if (existingBCIds.isEmpty()) {
+                    throw new IllegalStateException(
+                            "useExistingBC is true, but there is no BC to use.");
+                }
+                project.setExistingBCs(existingBCIds);
```

Comment:
In the finalizer, the ProjectHiearchy/ProjectDetail should be only read. Here, the field existingBCs should be already set because it was received from user.